lib/main.js

```diff
@@ -82,10 +84,15 @@ function tokenExchangeHandler({
 
       const defaultMaxAge = defaults.defaultMaxAge || DEFAULT_MAX_AGE;
 
-      const {accessToken, expiresIn, scope} = await issue({
+      const key = JSON.stringify([client, scopeRequested, body, defaultMaxAge]);
```

Comment:
If a client changes their secret then any old access token will continue to be handed out until the cache releases it (there is no forced cache clearing). We should document this behavior both in the changelog and in the docs for the function under the `cache` param.

Reply:
@dlongley - might that be an argument for including the client secret in the cache key?

Reply:
Added documentation, although I also thinking adding the client secret to the cache would work.

Reply:
Well, we wouldn't want to put the secret in the cache -- but a hash of it would be ok if we decide to take that direction in the future (something to keep in mind).

Reply:
(Yeah, I meant hash of it)